src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs

```diff
@@ -215,26 +216,30 @@ internal static unsafe class EETypeCreator
                 }
                 else if (state.TypeBeingBuilt.IsGenericDefinition)
                 {
-                    flags = (ushort)EETypeKind.GenericTypeDefEEType;
-                    isValueType = state.TypeBeingBuilt.IsValueType;
                     flags = EETypeBuilderHelpers.ComputeFlags(state.TypeBeingBuilt);
+                    Debug.Assert((flags & (uint)EETypeFlags.HasComponentSizeFlag) != 0);
+                    flags |= checked((ushort)state.TypeBeingBuilt.Instantiation.Length);
+
+                    isValueType = state.TypeBeingBuilt.IsValueType;
                     hasFinalizer = false;
                     isArray = false;
                     isNullable = false;
                     isGeneric = false;
                     isGenericEETypeDef = true;
                     isAbstractClass = false;
                     isByRefLike = false;
-                    componentSize = checked((ushort)state.TypeBeingBuilt.Instantiation.Length);
                     baseSize = 0;
                     typeManager = PermanentAllocatedMemoryBlobs.GetPointerToIntPtr(moduleInfo.Handle.GetIntPtrUNSAFE());
                 }
                 else
                 {
+                    flags = EETypeBuilderHelpers.ComputeFlags(state.TypeBeingBuilt);
```

Comment:
```suggestion
#if FEATURE_UNIVERSAL_GENERICS
                    flags = EETypeBuilderHelpers.ComputeFlags(state.TypeBeingBuilt);                 
```
and also add this to the end of the block
```
#else
    Debug.Assert(false);
#endif
```    
This whole block should be unreachable. If it is reachable, we would want to add some targeted tests.